src/main/java/uk/gov/cshr/config/CustomAuthenticationSuccessHandler.java

```diff
@@ -0,0 +1,117 @@
+package uk.gov.cshr.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.RedirectStrategy;
+import org.springframework.security.web.WebAttributes;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import uk.gov.cshr.domain.Identity;
+import uk.gov.cshr.repository.IdentityRepository;
+import uk.gov.cshr.service.AgencyTokenService;
+import uk.gov.cshr.service.EmailUpdateService;
+import uk.gov.cshr.service.security.IdentityDetails;
+import uk.gov.cshr.service.security.IdentityService;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.util.Optional;
+
+@Slf4j
+public class CustomAuthenticationSuccessHandler
+        implements AuthenticationSuccessHandler {
+
+    @Value("${emailUpdate.invalidDomainUrl}")
+    private String invalidDomainUrl;
+
+    @Value("${lpg.uiUrl}")
+    private String lpgUiUrl;
+
+    @Autowired
+    private RedirectStrategy redirectStrategy;
+
+    @Autowired
+    private AgencyTokenService agencyTokenService;
+
+    @Autowired
+    private IdentityService identityService;
+
+    @Autowired
+    private EmailUpdateService emailUpdateService;
+
+    @Autowired
+    private IdentityRepository identityRepository;
+
+    @Override
+    public void onAuthenticationSuccess(HttpServletRequest request,
+                                        HttpServletResponse response, Authentication authentication)
+            throws IOException {
+        handle(request, response, authentication);
+        clearAuthenticationAttributes(request);
+    }
+
+    protected void handle(HttpServletRequest request,
+                          HttpServletResponse response, Authentication authentication)
+            throws IOException {
+
+        String targetUrl = determineTargetUrl(authentication);
+
+        if (response.isCommitted()) {
+            log.debug(
+                    "Response has already been committed. Unable to redirect to "
+                            + targetUrl);
+            return;
+        }
+
+        log.debug("Redirecting to " + targetUrl);
+        redirectStrategy.sendRedirect(request, response, targetUrl);
+    }
+
+    protected String determineTargetUrl(Authentication authentication) {
+        String targetURL = lpgUiUrl;
```

Comment:
This seems to be redundant as the `lpgUIUrl` default is used in the `workoutIfUserShouldUseAnAgencyToken()` method

Reply:
Updated